### PR TITLE
sdl/all: bump deps

### PR DESCRIPTION
**Pull request by ericLemanissier**
Generated and committed by [Conan Center Bump Deps](https://github.com/ericLemanissier/conan-center-index-bump-deps) Find more updatable recipes in the [GitHub Pages](https://ericLemanissier.github.io/conan-center-index-bump-deps/)

Specify library name and version:  **lib/1.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
blocked by:
- https://github.com/conan-io/conan-center-index/pull/22241

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7544585641/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7544585641/attempts/2)

**Review by Croydon — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@uilianries this is a call for review

File: recipes/sdl/all/conanfile.py [hunk 1]
```diff
@@ -183,13 +179,7 @@ def package_id(self):
...
 
     def build_requirements(self):
-        if self.settings.os == "Macos" and cross_building(self):
-            # Workaround for CMake bug with error message:
-            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
-            # set. This could be because you are using a Mac OS X version less than 10.5
-            # or because CMake's platform configuration is corrupt.
-            # FIXME: Remove once CMake on macOS/M1 CI runners is upgraded.
-            self.tool_requires("cmake/3.27.9")
+        self.tool_requires("cmake/[>3.27 <4]")
```
Comment (uilianries) [inline recipes/sdl/all/conanfile.py:182]: This error still persists? We have update Mac machines to M2, maybe we have a different result now.

Comment (ericLemanissier) [reply recipes/sdl/all/conanfile.py:182]: somehow it failed without this tool_requires: https://c3i.jfrog.io/artifactory/misc-v2/logs/pr/22246/8-macos-clang/sdl/2.28.5/cde1046a936a59525b593ea75621a3b236a649fa-build.txt

**Review by uilianries — APPROVED** [review]

**Pull request merged**